src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -343,6 +344,7 @@ export class DBTCoreProjectIntegration
             adapter: this.getAdapterType() || "unknown", // TODO: this should be moved to dbtProject
           },
         );
+        this.isInitialized = true;
```

Comment:
maybe rename to `hasManifest` as this code will be hit a lot more, also what happens when project config is changed. Then manifest and config are erased but i don't see any code handling this.

Reply:
I will add a check, if `hasManifest` is true, then we don't `rebuildManifestDiagnostics` during `rebuildManifest`